src/main/kotlin/org/rust/ide/console/RsConsoleRunner.kt

```diff
@@ -62,17 +69,80 @@ class RsConsoleRunner(project: Project) :
         return consoleView
     }
 
-    override fun fillToolBarActions(
-        toolbarActions: DefaultActionGroup,
+
+
+    override fun createContentDescriptorAndActions() {
+        val defaultExecutor = executor
+        val runToolbarActions = DefaultActionGroup()
+        val runActionsToolbar = ActionManager.getInstance().createActionToolbar("RustConsoleRunner", runToolbarActions, false)
+
+        val outputToolbarActions = DefaultActionGroup()
+        val outputActionsToolbar = ActionManager.getInstance().createActionToolbar("RustConsoleRunner", outputToolbarActions, false)
+
+        val actionsPanel = JPanel(BorderLayout())
+        // Left toolbar panel
+        actionsPanel.add(runActionsToolbar.component, BorderLayout.WEST)
+        // Add line between toolbar panels
+        val outputActionsComponent = outputActionsToolbar.component
+        val emptyBorderSize = outputActionsComponent.border.getBorderInsets(outputActionsComponent).left
+        outputActionsComponent.border = BorderFactory.createCompoundBorder(BorderFactory.createMatteBorder(0, 1, 0, 0, JBColor.border()), JBEmptyBorder(emptyBorderSize))
+        // Right toolbar panel
+        actionsPanel.add(outputActionsComponent, BorderLayout.CENTER)
+
+        // Runner creating
+        val mainPanel = JPanel(BorderLayout())
+        mainPanel.add(actionsPanel, BorderLayout.WEST)
+        mainPanel.add(consoleView.component, BorderLayout.CENTER)
+
+        runActionsToolbar.setTargetComponent(mainPanel)
+        outputActionsToolbar.setTargetComponent(mainPanel)
+
+        val contentDescriptor = RunContentDescriptor(consoleView, processHandler, mainPanel, constructConsoleTitle(consoleTitle), consoleIcon)
+        Disposer.register(project, contentDescriptor)
+        contentDescriptor.setFocusComputable { consoleView.consoleEditor.contentComponent }
+        contentDescriptor.isAutoFocusContent = isAutoFocusContent
+
+        // tool bar actions
+        val actions: MutableList<AnAction> = fillRunActionsToolbar(defaultExecutor, runToolbarActions, contentDescriptor)
+        val outputActions: List<AnAction> = fillOutputActionsToolbar(outputToolbarActions)
+        actions.addAll(outputActions)
+
+        registerActionShortcuts(actions, consoleView.consoleEditor.component)
+        registerActionShortcuts(actions, mainPanel)
+
+        showConsole(defaultExecutor, contentDescriptor)
+    }
+
+    private fun fillRunActionsToolbar(
```

Comment:
I don't like this method as well as `fillOutputActionsToolbar`. See the code above on how it can be implemented.